### PR TITLE
Add create program page

### PR DESCRIPTION
**Pull request by d8660091**

### CODE REVIEW

File: client/pages/login.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ export default createPage({
     <div>
       <GoogleLogin
         id="google-login"
-        link={urlJoin(EGO_API_ROOT, `api/oauth/login/google?client_id=${EGO_CLIENT_ID}`)}
```
Comment (hlminh2000) [inline client/pages/login.js]: no need to change this, it'd be a breaking change for deployment. Just put the ego domain in your local for development? :)

File: client/pages/create-program.js [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+        </Col>
+      </Row>
+      <Row
+        css={css`
+          flex-direction: row-reverse;
+        `}
+      >
+        <Button onClick={() => {}}>Create</Button>
+      </Row>
+    </Container>
```
Comment (hlminh2000) [inline client/pages/create-program.js]: if you update against master, there should be a `programs` page set up that can be used as a boiler plate so we don't end up with a huge file for every page. We can refactor down the road to pull the common layouts into its own thing down the road :)

File: client/uikit/form/Input/index.js [hunk 3]
```diff
@@ -17,40 +18,40 @@ const Input = ({
...
-          placeholder={disabled ? '' : placeholder}
-          value={value}
-          type={type}
-          onChange={onChange}
-          size={size}
-          disabled={disabled}
-        />
-      </StyledInputWrapper>
-      {error && errorMessage ? <ErrorMsg>{errorMessage}</ErrorMsg> : null}
+      <FormControlContext.Consumer>
```
Comment (hlminh2000) [inline client/uikit/form/Input/index.js]: We'd prefer the `useContext` API now :)

File: client/uikit/form/Input/index.js [hunk 4]
```diff
@@ -17,40 +18,40 @@ const Input = ({
...
+        {contextValue => (
+          <StyledInputWrapper
+            size={size}
+            onFocus={() => setActive('focus')}
+            onBlur={() => setActive('default')}
+            error={contextValue.error}
+            disabled={contextValue.disabled}
```
Comment (hlminh2000) [inline client/uikit/form/Input/index.js]: Ideally I'd like to have inputs remain something that's usable without the context, so maybe we can have it fallback to the props :)

File: client/components/NavBar.js [hunk 5]
```diff
@@ -24,7 +24,7 @@ const NavbarLink = ({ path, active }: { path: string, active: boolean }) => {
   };
   return (
     <MenuItem
-      DomComponent={props => (
+      DomComponent={({ active, ...props }) => (
```
Comment (hlminh2000) [inline client/components/NavBar.js]: 👍

File: client/uikit/form/Input/index.js (hunk 3 above)
Comment (d8660091) [reply client/uikit/form/Input/index.js]: nice

File: client/uikit/form/Input/index.js [hunk 6]
```diff
@@ -17,40 +16,43 @@ const Input = ({
...
   className,
+  error,
+  disabled,
   ...props
 }) => {
   const [activeState, setActive] = useState('default');
+
+  const { disabled: calcDisabled = disabled, error: calcError = error } = useContext(
+    FormControlContext,
+  );
```
Comment (hlminh2000) [inline client/uikit/form/Input/index.js]: do we need `useContext(FormControlContext) || {}`? what happens when there's no context?

Comment (d8660091) [reply client/uikit/form/Input/index.js]: good catch

**Pull request merged**